### PR TITLE
[ZEPPELIN-1409] Refactor RAT build on Travis.CI configuration

### PR DESCRIPTION
**Pull request by lresende**
### What is this PR for?
Create a specific build for checking license compliance with RAT
and avoid running these checks on every build that compose the
PR build. 
As for normal development builds, this follows  the same pattern 
used for maven tests, RAT is enabled to run by default, but now
there is support disabling it with -DskipRat. 
Travis CI will run RAT once, on the RAT build, and disable RAT 
checks on all other build profiles.

### What type of PR is it?
[Enhancement]

### What is the Jira issue?
* [ZEPPELIN-1409](https://issues.apache.org/jira/browse/ZEPPELIN-1409)

### CODE REVIEW

**Comment by lresende** [comment]
One of the builds failed with the following, which seems not related

- should provide onclick method *** FAILED ***
  The code passed to eventually never returned normally. Attempted 1 times over 1.164197794 seconds. Last failure message: 0 was not equal to 1. (AbstractAngularElemTest.scala:72)

**Comment by lresende** [comment]
@bzz @Leemoonsoo Please review

File: .travis.yml [hunk 1]
```diff
@@ -36,6 +36,10 @@ matrix:
   include:
     # Test all modules with spark 2.0.0 and scala 2.11
```
Comment (Leemoonsoo) [inline .travis.yml]: Thanks for the contribution. Shell we have better comment here?

Comment (lresende) [inline .travis.yml]: @Leemoonsoo updated.

File: zeppelin-zengine/pom.xml [hunk 2]
```diff
@@ -129,12 +129,6 @@
     </dependency>
 
     <dependency>
-      <groupId>org.eclipse.jetty.websocket</groupId>
```
Comment (bzz) [inline zeppelin-zengine/pom.xml]: Is this related to RAT?

**Comment by bzz** [comment]
I think having a separate profile it is reasonable approach for CI. Though I also think there is value in keepinv the default behaviour for local Dev workflow and make RAT check on by default.

In order to communicate the value of such fix better, I would suggest  linking to JIRA/here the logs of CI builds that have faild due to RAT issue as well.

**Comment by lresende** [comment]
@bzz  this has now been updated to follow the same pattern used for maven tests, enable RAT to run by default, but support disabling it with -DskipRat. Travis CI will run RAT once, on the RAT build, and disable RAT checks on all other build profiles.

**Comment by corneadoug** [comment]
Then how about we use -DskipRat on all building profiles, and have a profile that run only RAT without building?

Build time is pretty long, so it would be nice to separate some easy check that do not need compilation or specific settings separated, we would keep the CI feedback loop shorter and it would also be easier to know what failed without going to the log.

It could probably also be applied to things like Style Check or Unit Tests later

**Comment by lresende** [comment]
@corneadoug This is exactly what is implemented on this pr.

**Comment by corneadoug** [comment]
@lresende My bad, when I first read I understood that RAT would be included in one profile that would still build everything. (just like we do with the -DskipTests)

**Comment by felixcheung** [comment]
LGTM

**Comment by bzz** [comment]
Looks good to me, merging if there is no further discussion.

CI fails on single profile with 1 flaky test

```
[INFO] Zeppelin: Server ................................... FAILURE [03:39 min]

Failed tests: 
  ZeppelinSparkClusterTest.zRunTest:204 expected:<FINISHED> but was:<ERROR>
```
